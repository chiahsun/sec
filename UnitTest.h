#ifndef UNIT_TEST_H
#define UNIT_TEST_H

#include <iostream>

#define assertEqual(a, b, non_equal_message) do { if ((a) != (b)) { std::cerr << "TestFail : (" << __FILE__ << " : " << __LINE__ << ")\nGolden: "\
    << (a) << "\nOutput: " << (b) << "\nMessage: " << non_equal_message << "\n"; }} while(0)

#endif // UNIT_TEST_H
