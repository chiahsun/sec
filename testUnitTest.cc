#include "UnitTest.h"

void testBool() {
    assertEqual(true, true, "true == true");
    assertEqual(true, false, "This should print this message since true != false");
}

int main() {
    testBool();
    return 0;
}
