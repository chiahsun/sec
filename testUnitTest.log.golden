TestFail : (testUnitTest.cc : 5)
Golden: 1
Output: 0
Message: This should print this message since true != false
